django/db/models/base.py

```diff
@@ -749,7 +755,10 @@ def save_base(self, raw=False, force_insert=False,
                 sender=origin, instance=self, raw=raw, using=using,
                 update_fields=update_fields,
             )
-        with transaction.atomic(using=using, savepoint=False):
+        parents = meta.parents.items()
+        # We do not need transaction if we are sure we are only going to issue __one__ query
+        context_manager = noop_cm() if (force_insert or force_update or update_fields) and not parents else transaction.atomic(using=using, savepoint=False)
```

Comment:
this would be better as a multiline if

Reply:
Yes, it is wip anyways -- I am more after a discussion about what the if should check for.